devlib/instrument/__init__.py

```diff
@@ -258,24 +258,29 @@ def teardown(self):
         pass
 
     def reset(self, sites=None, kinds=None, channels=None):
+        self.active_channels = []
         if kinds is None and sites is None and channels is None:
             self.active_channels = sorted(self.channels.values(), key=lambda x: x.label)
-        else:
-            if isinstance(sites, basestring):
-                sites = [sites]
-            if isinstance(kinds, basestring):
-                kinds = [kinds]
-            self.active_channels = []
-            for chan_name in (channels or []):
+        elif channels is not None:
+            if sites is not None or kinds is not None:
+                raise ValueError(
+                    'sites and kinds should not be set if channels is set')
```

Comment:
nitpick: I'd put the closing parenthesis on a new line.

Reply:
Just did `git --no-pager grep -n -e \(\$ -- ` to find other instances, looks like the rest of the code closes it on the same line